logstash-core/src/main/java/org/logstash/execution/ShutdownWatcherExt.java

```diff
@@ -105,29 +104,35 @@ public IRubyObject shutdownStalled(final ThreadContext context) {
         if (reports.size() != reportEvery) {
             return context.fals;
         }
-        final int[] inflightCounts = reports.stream().mapToInt(
-            obj -> obj.callMethod(context, "inflight_count").convertToInteger().getIntValue()
-        ).toArray();
-        boolean stalled = true;
+
+        final int[] inflightCounts = reports.stream()
+                .mapToInt(obj -> obj.callMethod(context, "inflight_count").convertToInteger().getIntValue())
+                .toArray();
+
         for (int i = 0; i < inflightCounts.length - 1; ++i) {
             if (inflightCounts[i] > inflightCounts[i + 1]) {
-                stalled = false;
-                break;
+                return context.fals;
             }
         }
-        if (stalled) {
-            final IRubyObject[] stallingThreads = reports.stream().map(
-                obj -> obj.callMethod(context, "stalling_threads")
-            ).toArray(IRubyObject[]::new);
-            for (int i = 0; i < stallingThreads.length - 1; ++i) {
-                if (!stallingThreads[i].op_equal(context, stallingThreads[i + 1]).isTrue()) {
-                    stalled = false;
-                    break;
-                }
+
+        final IRubyObject[] stallingThreads = reports.stream()
+                .map(obj -> obj.callMethod(context, "stalling_threads"))
+                .toArray(IRubyObject[]::new);
+
+        if (inflightCounts.length == 0 && stallingThreads.length == 0) {
+            return context.fals;
+        }
```

Comment:
Given that these are derived from the reports array which is guaranteed to be of size `reportEvery`, these arrays should also be of size `reportEvery` (5)